src/seismometer/_api.py

```diff
@@ -176,86 +183,126 @@ def fairness_audit(metric_list: Optional[list[str]] = None, fairness_threshold=1
+        return template.render_censored_plot_message(sg.censor_threshold)
+    fairness_audit_to_html(
+        fairness_path, data, cohort_columns, score_column, target, score_threshold, metric_list, fairness_threshold
     )
+    return load_as_iframe(fairness_path, height=height)
 
-    def on_widget_value_changed(*args):
-        sg.selected_cohort = widget.value
 
-    widget.observe(on_widget_value_changed, "value")
+# endregion
 
-    # get intial value
-    on_widget_value_changed()
-    return widget
+# region notebook IPWidgets # TODO - move to .controls submodule
 
 
 @export
-@display_cached_widget
 def cohort_list():
     """
     Displays an exhaustive list of available cohorts for analysis.
     """
     sg = Seismogram()
-    from ipywidgets import HBox, Output
+    from ipywidgets import Output, VBox
 
     from .controls.selection import MultiSelectionListWidget
-    from .data.filter import filter_rule_from_cohort_dictionary
 
     options = sg.available_cohort_groups
 
-    comparison_selections = MultiSelectionListWidget(
-        options, ghost_text="Select cohort groups", title="Available Cohorts"
-    )
+    comparison_selections = MultiSelectionListWidget(options, title="Cohort")
     output = Output()
 
     def on_widget_value_changed(*args):
-        output.clear_output()
         with output:
-            if comparison_selections.value:
-                rule = filter_rule_from_cohort_dictionary(comparison_selections.value)
-                cohort_count = sum(rule.mask(sg.dataframe))
-                if cohort_count < 10:
-                    print("Selected Cohort is empty, or has fewer than 10 predictions.")
-                else:
-                    print(f"Selected Cohort has {sum(rule.mask(sg.dataframe))} predictions.")
-            else:
-                print(f"No Cohort Selected: {len(sg.dataframe)} predictions total.")
+            html = _cohort_list_details(comparison_selections.value)
+            output.clear_output()
+            display(html)
 
     comparison_selections.observe(on_widget_value_changed, "value")
 
-    # get intial value
+    # get initial value
     on_widget_value_changed()
 
-    return HBox(children=[comparison_selections, output])
+    return VBox(children=[comparison_selections, output])
+
+
+@disk_cached_html_segment
+def _cohort_list_details(cohort_dict: dict[str, tuple]) -> HTML:
+    from .data.filter import filter_rule_from_cohort_dictionary
+
+    sg = Seismogram()
+    rule = filter_rule_from_cohort_dictionary(cohort_dict)
+    data = rule.filter(sg.dataframe)
+    cohort_count = data[sg.entity_keys[0]].nunique()
+    if cohort_count < sg.censor_threshold:
+        return template.render_censored_plot_message(sg.censor_threshold)
+
+    cfg = sg.config
+    target_cols = [pdh.event_value(x) for x in cfg.targets]
+    intervention_cols = [pdh.event_value(x) for x in cfg.interventions]
+    outcome_cols = [pdh.event_value(x) for x in cfg.outcomes]
```

Comment:
in pandas_helpers: event_value, event_time, event_name: Pylance is saying the first `return None` is unreachable, because the typehints say that event must be a str.

`def event_value(event: Optional[str]) -> Optional[str]:` would probably be more accurate

Reply:
Instead removed the None check, since the event is typed to string.